### PR TITLE
Add check to verify the source instance template os linux distro

### PR DESCRIPTION
**Pull request by williamaronli**
Add check to verify the source instance template os linux distro

### CODE REVIEW

File: scripts/asm-installer/asm_vm [hunk 1]
```diff
@@ -1110,6 +1110,22 @@ this project, run:
   --project="${PROJECT_ID}"
 EOF
     fi
+
+    local INSTANCE_TEMPLATE_REQ=$(retry 2 run gcloud compute instance-templates \
+      describe "${SOURCE_INSTANCE_TEMPLATE}" \
```
Comment (JimmyCYJ) [inline scripts/asm-installer/asm_vm:1115]: What if user does not specify SOURCE_INSTANCE_TEMPLATE? For example, user just want to create debian VM with the default setup.

`INSTANCE_TEMPLATE_REQ` is not accurate, this is more about getting SOURCE_INSTANCE_TEMPLATE, it is not relevant to any requirement.

File: scripts/asm-installer/asm_vm [hunk 2]
```diff
@@ -1110,6 +1110,22 @@ this project, run:
...
+      --format=json \
+      --project="${PROJECT_ID}")
+    local source_image_path="$(echo "${INSTANCE_TEMPLATE_REQ}" \
+    | jq '.properties.disks[].initializeParams.sourceImage')"
+
+    # check whether it is a  public gcp image
```
Comment (JimmyCYJ) [inline scripts/asm-installer/asm_vm:1121]: There are multiple spaces between "a  public". public is not accurate here. 
This is checking if the chosen image is supported. There are other images available for GCE VM, which are public but not supported.

File: scripts/asm-installer/asm_vm [hunk 3]
```diff
@@ -1110,6 +1110,22 @@ this project, run:
...
+    if [[ "$source_image_path" == *projects* && "$source_image_path" == *global/images/* ]]; then
+      if [[ "$source_image_path" != *debian-9* ]] && [[ "$source_image_path" != *debian-10* ]]; then
+        fatal "found not support image, the image path is $source_image_path"
```
Comment (JimmyCYJ) [inline scripts/asm-installer/asm_vm:1124]: @jasonwzm are there any test files that should be used for adding tests to cover this change?

File: scripts/asm-installer/asm_vm [hunk 4]
```diff
@@ -1110,6 +1110,22 @@ this project, run:
...
+
+    local INSTANCE_TEMPLATE_REQ=$(retry 2 run gcloud compute instance-templates \
+      describe "${SOURCE_INSTANCE_TEMPLATE}" \
+      --format=json \
+      --project="${PROJECT_ID}")
+    local source_image_path="$(echo "${INSTANCE_TEMPLATE_REQ}" \
+    | jq '.properties.disks[].initializeParams.sourceImage')"
+
+    # check whether it is a  public gcp image
+    if [[ "$source_image_path" == *projects* && "$source_image_path" == *global/images/* ]]; then
```
Comment (jasonwzm) [inline scripts/asm-installer/asm_vm:1122]: We need to check the full path, e.g. projects/[debian|centos]-cloud/global/images (syntax may not be correct)

File: scripts/asm-installer/asm_vm (hunk 3 above)
Comment (jasonwzm) [inline scripts/asm-installer/asm_vm:1124]: Should have something like: "Image not supported. Supported images are..."

File: scripts/asm-installer/asm_vm [hunk 5]
```diff
@@ -1110,6 +1110,22 @@ this project, run:
...
+      if [[ "$source_image_path" != *debian-9* ]] && [[ "$source_image_path" != *debian-10* ]]; then
+        fatal "found not support image, the image path is $source_image_path"
+      fi
+    else
+      info "custom image found, skip checking"
```
Comment (jasonwzm) [inline scripts/asm-installer/asm_vm:1127]: If you output a skip message here, you should also have a message before this saying "Verifying GCE image..."

File: scripts/asm-installer/asm_vm (hunk 3 above)
Comment (jasonwzm) [reply scripts/asm-installer/asm_vm:1124]: Yes. There are two test files in /tests folder that cover VM.

File: scripts/asm-installer/asm_vm (hunk 1 above)
Comment (jasonwzm) [reply scripts/asm-installer/asm_vm:1115]: Just use single local variable should be enough, as you are only verifying the image path in this function. Please capitalize the local variables as well to be consistent with other local variables.

File: scripts/asm-installer/asm_vm [hunk 6]
```diff
@@ -1110,6 +1110,22 @@ this project, run:
   --project="${PROJECT_ID}"
 EOF
     fi
+
+    local INSTANCE_TEMPLATE_REQ=$(retry 2 run gcloud compute instance-templates \
+      describe "${SOURCE_INSTANCE_TEMPLATE}" \
+      --format=json \
+      --project="${PROJECT_ID}")
+    local source_image_path="$(echo "${INSTANCE_TEMPLATE_REQ}" \
+    | jq '.properties.disks[].initializeParams.sourceImage')"
```
Comment (jasonwzm) [inline scripts/asm-installer/asm_vm:1119]: What if there are more than one disks for this instance?

File: scripts/asm-installer/asm_vm (hunk 3 above)
Comment (jasonwzm) [reply scripts/asm-installer/asm_vm:1124]: Please also add test in the same PR.

**Review by jasonwzm — CHANGES_REQUESTED** [review]

**Comment by williamaronli** [comment]
Create a new one and close this one will address the comments in this PR: https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/pull/541

**Pull request closed without merging**